### PR TITLE
processDiagnosticMessages.ts: linted, removed unused code

### PR DESCRIPTION
**Pull request by NaridaL**
Added following line to generated output:
"// generated from 'src/diagnosticMessages.json' by 'scripts/processDiagnosticMessages.ts'\r\n" +

Fixes https://github.com/Microsoft/TypeScript/issues/3591

### CODE REVIEW

File: scripts/processDiagnosticMessages.ts [hunk 1]
```diff
@@ -9,104 +10,102 @@ interface DiagnosticDetails {
...
-            throw new Error(`Name ${name} appears twice`);
-        seenNames.set(name, true);
-    }
+    const diagnosticMessagesJson: { [key: string]: DiagnosticDetails } = JSON.parse(inputStr);
 
     const diagnosticMessages: InputDiagnosticMessageTable = ts.createMapFromTemplate(diagnosticMessagesJson);
 
-    var infoFileOutput = buildInfoFileOutput(diagnosticMessages);
+    const outputFilesDir = ts.getDirectoryPath(inputFilePath);
+    const thisFilePathRel = ts.getRelativePathToDirectoryOrUrl(outputFilesDir, sys.getExecutingFilePath(),
```
Comment (NaridaL) [inline scripts/processDiagnosticMessages.ts]: overkill?

File: scripts/processDiagnosticMessages.ts [hunk 2]
```diff
@@ -9,57 +10,55 @@ interface DiagnosticDetails {
...
     }
 
-    var inputFilePath = sys.args[0].replace(/\\/g, "/");
-    var inputStr = sys.readFile(inputFilePath);
+    const inputFilePath = sys.args[0].replace(/\\/g, "/");
+    const inputStr = sys.readFile(inputFilePath);
 
-    var diagnosticMessagesJson: { [key: string]: DiagnosticDetails } = JSON.parse(inputStr);
-    // Check that there are no duplicates.
-    const seenNames = ts.createMap<true>();
```
Comment (mhegazy) [inline scripts/processDiagnosticMessages.ts]: why was this removed?

File: scripts/processDiagnosticMessages.ts (hunk 1 above)
Comment (mhegazy) [reply scripts/processDiagnosticMessages.ts]: yeah.. i would say so.

Comment (mhegazy) [reply scripts/processDiagnosticMessages.ts]: but i can live with it :)

File: scripts/processDiagnosticMessages.ts (hunk 2 above)
Comment (NaridaL) [reply scripts/processDiagnosticMessages.ts]: Object.keys never returns duplicates, the whole block was useless.

Comment (mhegazy) [reply scripts/processDiagnosticMessages.ts]: :D nice catch.
